lib/dash/segment_template.js

```diff
@@ -702,6 +702,7 @@ shaka.dash.SegmentTemplate = class {
         encrypted);
     ref.codecs = context.representation.codecs;
     ref.mimeType = context.representation.mimeType;
+    ref.boundaryEnd = context.periodInfo.start + context.periodInfo.duration;
```

Comment:
So right now it only works for `SegmentTemplate`?

Reply:
Yes, I'd like to eventually populate `boundaryEnd` based on HLS discontinuities too but I think we might take that up in a different issue if that's OK.